### DISCUSSION TITLE
Parse markdown, apply changes to text, get markdown back

### DISCUSSION THREAD

**Discussion by nschloe** [executablebooks/markdown-it-py#306] [Q&A]
I'd like to parse a Markdown file, do some simple replacement in "text" (not code blocks, not link targets), and get the otherwise unchanged Markdown back. Is markdown-it-py suitable for this? The `mdformat` renderer seems to do some changes, e.g.:

```python
from markdown_it import MarkdownIt
from mdformat.renderer import MDRenderer

md = MarkdownIt("commonmark")

source_markdown = """
Here's some *text* and a footnote [^a]

1. a list

> a *quote*

[^a]: A
"""

tokens = md.parse(source_markdown)

renderer = MDRenderer()
output_markdown = renderer.render(tokens, options={}, env={})

print(output_markdown)
```
```
Here's some *text* and a footnote [^a](A)

1. a list

> a *quote*
```

**Comment by 4l1fe** [comment]
Join to the question. My need is to do a few modifications on source text and get raw markdown back to convey it to the next stage in a pipeline.

**Comment by 4l1fe** [comment]
@nschloe i guess it is what we are looking for https://github.com/miyuchina/mistletoe

> Markdown (Can be used to reflow the text, or make other types of automated changes to Markdown documents)

[markdown-to-markdown](https://github.com/miyuchina/mistletoe/blob/master/dev-guide.md#markdown-to-markdown-parsing-and-rendering)
